.github/workflows/open_caller_prs.py

```diff
@@ -0,0 +1,178 @@
+- clones the target repository into a temporary directory, and cd into it
+- writes the source directory's template_workflow_file into the target's /.github/workflows directory
+- git checkout, add, commit and push to the target's remote 
+- opens a PR against origin/main
+- requests a review from "scala-reviewers" and "js-ts-reviewers"
+- once this is all done, moves back to the source directory and removes the temporary directory
+"""
+
+import sys
+import subprocess
+import contextlib
+import os
+import re
+import shutil
+import tempfile
+import boto3
+import httpx
+
+def _subprocess_run(cmd, exit_on_error=True):
+    print("*** Running %r" % " ".join(cmd))
+
+    output = []
+    pipe = subprocess.Popen(
+        cmd, encoding="utf8", stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+    )
+
+    # Await command completion and print lines as they come in
+    for stdout_line in iter(pipe.stdout.readline, ""):
+        print(stdout_line, end="")
+        output.append(stdout_line)
+
+    # Extract results
+    pipe.communicate()
+    return_code = pipe.returncode
+
+    if return_code != 0 and exit_on_error:
+        sys.exit(return_code)
+
+    return "".join(output).strip()
+
+
+def git(*args, exit_on_error=True):
+    """Run a Git command and return its output."""
+    cmd = ["git"] + list(args)
+
+    return _subprocess_run(cmd, exit_on_error=exit_on_error)
+
+@contextlib.contextmanager
+def working_directory(target_dir):
+    """
+    Changes the working directory to the given path, then returns to the
+    original directory when done.
+    """
+    source_dir = os.getcwd()
+
+    os.chdir(target_dir)
+    try:
+        yield
+    finally:
+        os.chdir(source_dir)
+
+
+@contextlib.contextmanager
+def cloned_repo(git_url):
+    """
+    Clones the repository and changes the working directory to the cloned
+    repo.  Cleans up the clone when it's done.
+    """
+    target_dir = tempfile.mkdtemp()
+
+    git("clone", git_url, target_dir)
+
+    try:
+        with working_directory(target_dir):
+            yield
+    finally:
+        shutil.rmtree(target_dir)
+
+
+class AlreadyAtLatestVersionException(Exception):
+    pass
+
+
+def create_or_update_github_workflow(source_file_path):
+    with open(source_file_path, "r") as source_file, open(f".github/workflows/{template_workflow_file}", "w") as target_file:
+        for line in source_file:
+            target_file.write(line)
+
+def get_last_merged_pr_number():
+    for line in git("log", "--oneline").splitlines():
+        m = re.match(r"^[0-9a-f]{7} Merge pull request #(?P<pr_number>\d+)", line)
+
+        if m is not None:
+            return m.group("pr_number")
+
+def get_github_api_key():
+    session = boto3.Session()
+    secrets_client = session.client("secretsmanager")
+    # using the scala_libs_pr_bumps secret for now but this should have its own secret
+    secret_value = secrets_client.get_secret_value(SecretId="builds/github_wecobot/scala_libs_pr_bumps")
```

Comment:
We should avoid further use of GitHub API keys for weco-bot. Is the intention to automate running this script?